pdb-alignment-web/src/main/java/org/cbioportal/pdb_annotation/web/controllers/MainGetAlignmentsController.java

```diff
@@ -0,0 +1,323 @@
+                // P04637
+                outList.addAll(seqController.getPdbAlignmentByUniprotAccession(id));
+
+            } else if (id.split("_").length == 2) {// ID: P53_HUMAN
+                outList.addAll(seqController.getPdbAlignmentByUniprotId(id));
+            } else {
+                log.info("Error in Input. id_type:Uniprot id: " + id);
+            }
+        } else if (id_type.equals("uniprot_isoform")) {
+            if (id.split("_").length == 2 && id.split("_")[0].length() == 6) {// Accession:
+                // P04637
+                outList.addAll(seqController.getPdbAlignmentByUniprotAccessionIso(id.split("_")[0], id.split("_")[1]));
+
+            } else if (id.split("_").length == 3) {// ID: P53_HUMAN
+
+                outList.addAll(seqController.getPdbAlignmentByUniprotIdIso(id.split("_")[0] + "_" + id.split("_")[1],
+                        id.split("_")[2]));
+            } else {
+                log.info("Error in Input. id_type:Uniprot_isoform id: " + id);
+            }
+        } else {
+            log.info("Error in Input. id_type:" + id_type + " id: " + id);
+        }
+        return outList;
+    }
+
+    @RequestMapping(value = "/alignments/{id_type}/{id:.+}/pdb/{pdb_id}_{chain_id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation("Get PDB Alignments by ProteinId, PDBId and Chain")
+    public List<Alignment> getAlignmentByPDB(
+            @ApiParam(required = true, value = "Input id_type: ensembl; uniprot; uniprot_isoform") @PathVariable String id_type,
+            @ApiParam(required = true, value = "Input id e.g.\n"
+                    + "ensembl:ENSP00000484409.1/ENSG00000141510.16/ENST00000504290.5; "
+                    + "uniprot:P04637/P53_HUMAN; uniprot_isoform:P04637_9/P53_HUMAN_9 ") @PathVariable String id,
+            @ApiParam(required = true, value = "Input PDB Id e.g. 2fej") @PathVariable String pdb_id,
+            @ApiParam(required = true, value = "Input Chain e.g. A") @PathVariable String chain_id) {
+
+        List<Alignment> outList = new ArrayList<Alignment>();
+        if (id_type.equals("ensembl")) {
+
+            if (id.startsWith("ENSP")) {// EnsemblID:
+                // ENSP00000484409.1/ENSP00000484409
+
+                List<Ensembl> ensembllist = ensemblRepository.findByEnsemblIdStartingWith(id);
+                for (Ensembl ensembl : ensembllist) {
+
+                    List<Alignment> list = alignmentRepository.findBySeqId(ensembl.getSeqId());
+
+                    for (Alignment ali : list) {
+                        String pd = ali.getPdbId().toLowerCase();
+                        String ch = ali.getChain().toLowerCase();
+                        if (pd.equals(pdb_id.toLowerCase()) && ch.equals(chain_id.toLowerCase())) {
+                            outList.add(ali);
+                        }
+                    }
+
+                }
+
+            } else if (id.startsWith("ENSG")) {// EnsemblGene:
```

Comment:
double check this ... I was having trouble passing in ENSG genes (with or without a version number)